classes/phing/tasks/ext/dbdeploy/DbDeployTask.php

```diff
@@ -260,28 +260,31 @@ protected function generateSql($undo = false)
                     to avoid the delta failing.  You may need to manually undo part of this delta.\n\n'
                         . $contents, Project::MSG_WARN);
                 }
-                // allow construct with and without space added
-                $split = strpos($contents, '-- //@UNDO');
-                if ($split === false) {
-                    $split = strpos($contents, '--//@UNDO');
-                }
-                if ($split === false) {
-                    $split = strlen($contents);
+
+                // ignore tabs and spaces before @UNDO and any characters after in that line
+                $split = preg_split('/--[\t ]*\/\/@UNDO[^\r\n]*/', $contents);
+
+                if ($split === false){
+                    $split = array($contents);
                 }
 
+                $deploySql = $split[0];
+                $undoSql = isset($split[1]) ? $split[1] : '';
+
                 if ($undo) {
-                    $sql .= substr($contents, $split + 10) . "\n";
+                    $sql .= $undoSql;
+                    $sql .= PHP_EOL;
                     $sql .= 'DELETE FROM ' . DbDeployTask::$TABLE_NAME . '
 	                         WHERE change_number = ' . $fileChangeNumber . '
```

Comment:
There are tabs here, please expand/replace them to spaces according to Coding Standards. Thanks!